### PR TITLE
chore: cleanup and add comments to Config classes

### PR DESCRIPTION
**Pull request by nathan-weinberg**
**Issue resolved by this Pull Request:**
Resolves https://github.com/instructlab/instructlab/issues/1538

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if nessessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

**Review by danmcp — APPROVED** [review]

File: src/instructlab/configuration.py
Comment (khaledsulayman) [inline src/instructlab/configuration.py:458]: nit: not a huge fan of mentioning issues in code

**Review by khaledsulayman — APPROVED** [review]
Much needed change. Thanks for this!

Small nit but won't block on it. Ofc I'll approve again if you remove that line 😉

**Review by leseb — APPROVED** [review]

File: src/instructlab/configuration.py
Comment (nathan-weinberg) [reply src/instructlab/configuration.py:458]: Valid opinion, but this is done elsewhere in the codebase at this time - going to keep it for now

File: src/instructlab/configuration.py
Comment (RobotSail) [inline src/instructlab/configuration.py:554]: You don't need to mention that it's a class, since that's already inferred. You can simplify this (and others) to be just "Configuration of the 'x' sub command"

File: src/instructlab/configuration.py
Comment (RobotSail) [inline src/instructlab/configuration.py:675]: These comments seem redundant since you've already explained what they are in the above docstrings + all of the other docstrings for each class

**Review by RobotSail — APPROVED** [review]
a few comments but otherwise LGTM

File: src/instructlab/configuration.py
Comment (nathan-weinberg) [reply src/instructlab/configuration.py:675]: Very fair point!

File: src/instructlab/configuration.py
Comment (nathan-weinberg) [reply src/instructlab/configuration.py:554]: Merged before I had a chance to respond, but yeah was just making this consistent with everything else - lot of this will probably get reworked with @cdoern's config redesign

**Pull request merged**
